src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
-export interface Figure {
+type ShapeOfFigure = 'triangle' | 'circle' | 'rectangle';
+type ColorOfFigure = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: ShapeOfFigure;
+  color: ColorOfFigure;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: ShapeOfFigure = 'triangle';
 
-}
+  constructor(
+    public color: ColorOfFigure,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (Math.min(this.a, this.b, this.c) <= 0
+    || [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[2]
+    >= [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[0]
+    + [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[1]) {
+      throw new Error(`It’s even hard for me to imagine
+      what kind of triangle this should be.
+      I'm not a quantum computer, can I have a normal triangle?`);
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = 0.5 * (this.a + this.b + this.c);
 
+    return Math.floor(Math.sqrt(p * (p - this.a)
+    * (p - this.b) * (p - this.c)) * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?

Reply:
code style 🥲

Reply:
its haute couture